tools/database/src/memtrie.rs

```diff
@@ -78,3 +93,158 @@ impl LoadMemTrieCommand {
         Ok(())
     }
 }
+
+#[derive(clap::Parser)]
+pub struct SplitShardTrieCommand {
+    #[clap(long)]
+    shard_uid: ShardUId,
+    #[clap(long)]
+    boundary_account: AccountId,
+}
+
+impl SplitShardTrieCommand {
+    pub fn run(
+        &self,
+        home: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let env_filter = EnvFilterBuilder::from_env().verbose(Some("memtrie")).finish()?;
+        let _subscriber = default_subscriber(env_filter, &Default::default()).global();
+        let mut near_config = nearcore::config::load_config(&home, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+        near_config.config.store.load_memtries_for_tracked_shards = true;
+        let genesis_config = &near_config.genesis.config;
+
+        let rocksdb = Arc::new(open_rocksdb(home, near_store::Mode::ReadOnly)?);
+        let store = near_store::NodeStorage::new(rocksdb).get_hot_store();
+        let final_head = store.chain_store().final_head()?;
+
+        let epoch_manager =
+            EpochManager::new_arc_handle(store.clone(), &genesis_config, Some(home));
+        let old_shard_layout = epoch_manager.get_shard_layout(&final_head.epoch_id)?;
+
+        let runtime =
+            NightshadeRuntime::from_config(home, store.clone(), &near_config, epoch_manager)
+                .context("could not create the transaction runtime")?;
+
+        let shard_layout =
+            ShardLayout::derive_shard_layout(&old_shard_layout, self.boundary_account.clone());
+        let child_shards = shard_layout
+            .get_children_shards_uids(self.shard_uid.shard_id())
+            .ok_or_else(|| anyhow::anyhow!("Cannot get child shards"))?;
+        let left_child_shard = child_shards[0];
+        let right_child_shard = child_shards[1];
+
+        let epoch_manager = DummyEpochManager::new(shard_layout);
+        let resharding_handle = ReshardingHandle::new();
+        let mut resharding_config_override = near_config.config.resharding_config;
+        resharding_config_override.batch_size = ByteSize::tb(1);
+        let resharding_config = near_config.client_config.resharding_config.clone();
+        resharding_config.update(resharding_config_override);
+        let resharder =
+            FlatStorageResharder::new(epoch_manager, runtime, resharding_handle, resharding_config);
```

Comment:
You're splitting the flat storage which takes a few minutes. You can try splitting the memtrie instead, it should be instant.

Reply:
cc @Longarithm is there an easy way to do it? 

If there really isn't let's stick to flat storage, this tool will be used very rarely so it's fine if it's a bit slow.

Reply:
The only method I found was `ReshardingManager::split_shard`, but it is private and would require instantiating the `ReshardingManager` to be used outside. All the logic is spread out into multiple other methods, and I'm not sure if all of them are relevant to our use case, e.g. do we care about congestion info here? Last, but not least, the comment on `process_memtrie_resharding_storage_update` says:
>  /// Creates temporary memtries for new shards to be able to process them in the next epoch.
> /// Note this doesn't complete memtries resharding, proper memtries are to be created later.

After executing this, `split_shard` sends a message to the resharding actor.

Reply:
All right, let's keep it, let me know once you've tested it and I'll have a final look.